packages/js/components/src/analytics/error/tests/index.test.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { createElement } from '@wordpress/element';
+import { render, screen, fireEvent } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import AnalyticsError from '..';
+
+describe( 'AnalyticsError', () => {
+	function onError( event ) {
+		// Note: this will swallow reports about unhandled errors!
+		// Use with extreme caution.
+		event.preventDefault();
+	}
+
+	// Mock window.location.reload by using a global variable
+	const originalLocation = window.location;
+
+	beforeAll( () => {
+		// Opt Out of the jsdom error messages
+		window.addEventListener( 'error', onError );
+		delete window.location;
+		window.location = { reload: jest.fn() };
+	} );
+
+	afterAll( () => {
+		window.location = originalLocation;
+		window.removeEventListener( 'error', onError );
+	} );
+
+	it( 'displays an error message', () => {
+		render( <AnalyticsError /> );
+
+		expect(
+			screen.getByText(
+				'There was an error getting your stats. Please try again.'
+			)
+		).toBeInTheDocument();
+	} );
+
+	it( 'shows reload button', () => {
+		render( <AnalyticsError /> );
+
+		expect( screen.getByText( 'Reload' ) ).toBeInTheDocument();
```

Comment:
Would it be a bit better to query the button by role?

```suggestion
		expect(
			screen.getByRole( 'button', { name: 'Reload' } )
		).toBeInTheDocument();
```